utils/azureCredentialsManager.ts

```diff
@@ -0,0 +1,138 @@
+import { SubscriptionClient, ResourceManagementClient, SubscriptionModels } from 'azure-arm-resource';
+import { AzureAccount } from '../typings/azure-account.api';
+import { ServiceClientCredentials } from 'ms-rest';
+import { AsyncPool } from '../utils/asyncpool';
+import { ContainerRegistryManagementClient } from 'azure-arm-containerregistry';
+import * as ContainerModels from '../node_modules/azure-arm-containerregistry/lib/models';
+import { ResourceGroup, ResourceGroupListResult } from "azure-arm-resource/lib/resource/models";
+import { MAX_CONCURRENT_SUBSCRIPTON_REQUESTS } from './constants';
+
+/* Singleton for facilitating communication with Azure account services by providing extended shared
+  functionality and extension wide access to azureAccount. Tool for internal use.
+  Authors: Esteban Rey L, Jackson Stokes
+*/
+
+export class AzureCredentialsManager {
+
+    //SETUP
+    private static _instance: AzureCredentialsManager = new AzureCredentialsManager();
+    private azureAccount: AzureAccount;
+
+    constructor() {
+        if (AzureCredentialsManager._instance) {
+            throw new Error("Error: Instantiation failed: Use SingletonClass.getInstance() instead of new.");
+        }
+        AzureCredentialsManager._instance = this;
+    }
+
+    public static getInstance(): AzureCredentialsManager {
+        return AzureCredentialsManager._instance;
+    }
```

Comment:
Your code is correct but as a more general implementation of a Singleton pattern, you can make the `constructor` private instead of doing manual check and do a lazy initializtion of class instance in the `getInstance` method?

```
class Singleton {
    private static _instance: Singleton;
    private constructor() {
        // const make private to control object creation
    }
    public static getInstance() {
        if (!Singleton.instance) { // lazy initialization
            Singleton.instance = new Singleton();
            // ... any other one time initialization goes here ...
        }
        return Singleton.instance;
    }
}
```

Reply:
Thanks for the feedback, I should have done that honestly! . Ill update that by early tomorrow.